### ISSUE TITLE
internal docker ports not exposed

### ISSUE THREAD

**Issue by nhumrich** [wercker/support#139]
TL;DR
----
"ports" on the docker push step dont actually exposes the port I want to be exposed.

Full explanation
----
Trying to use the internal/docker-push step on ewok and am providing the ports to be exposed:

    ports: ["5000","8085"]

The ports successfully get opened, and I can connect to them from other containers when running these containers locally with dockers --link command.

    $ curl mycontainer:8085
    OK

However, the ports are not exposed, meaning the are not published to docker that they exist:

    $ docker ps
    CONTAINER ID        IMAGE                         COMMAND                  CREATED              STATUS             PORTS               NAMES
    de5f9ec55f72        somename/mycontainer         "java"                    4 seconds ago       Up 2 seconds                            brown_fox
    f361903ba1ae        postgresdbsql                "/docker-entrypoint.s"    4 seconds ago       Up 3 seconds        5432/tcp            lonely_nobel

Notice that postgres shows `5432/tcp` but mycontainer shows nothing on ports.

But if I build the container myself, the ports are exposed:
    
    CONTAINER ID        IMAGE                         COMMAND      CREATED             STATUS              PORTS                NAMES
    51ad29272cc8        somename/mycontainer          "java"       4 seconds ago       Up 3 seconds        5000/tcp, 8085/tcp   loving_kirch

Here is my Dockerfile:
    
    FROM baseimage
    EXPOSE 5000 8085
    ENTRYPOINT ["java","-jar","/app/service.jar"]

And my wercker step looks like:

    box: baseimage
    ....
      - internal/docker-push:
        repository: $REPO
        tag: latest
        username: $DOCKER_USERNAME
        password: $DOCKER_PASSWORD
        ports: ["5000","8085"]
        entrypoint: "java -jar /app/service.jar"

Normally, I would say, "I dont really care, its not that big of a deal since I can still connect to it", however, this causes other issues on wercker, probably because wercker uses the exposed ports to open up firewalls or something.

So, wercker typically exposes environment variables. For example, on postgres I get is:

    POSTGRESSQL_ENV_POSTGRES_PASSWORD=secret
    POSTGRESSQL_PORT=tcp://172.17.27.197:5432
    POSTGRESSQL_PORT_5432_TCP_PROTO=tcp
    POSTGRESSQL_NAME=/wercker-pipeline-5605902bce3c3b07460ff35d/postgresdb
    POSTGRESSQL_ENV_PG_VERSION=9.4.4-1.pgdg80+1
    POSTGRESSQL_ENV_PG_MAJOR=9.4
    POSTGRESSQL_PORT_5432_TCP_PORT=5432
    POSTGRESSQL_PORT_5432_TCP=tcp://172.17.27.197:5432
    POSTGRESSQL_PORT_5432_TCP_ADDR=172.17.27.197
    POSTGRESSQL_ENV_PGDATA=/var/lib/postgresql/data
    POSTGRESDB_ENV_LANG=en_US.utf8

But for mycontainer, all I get it: 

    MYCONTAINER_ENV_LANG=en_US.UTF-8
    MYCONTAINER_NAME=/wercker-pipeline-5605902bce3c3b07460ff35d/directory-service
    MYCONTAINER_ENV_JAVA_HOME=/usr/lib/jvm/java-8-oracle
    MYCONTAINER_ENV_LC_ALL=en_US.UTF-8

So I dont know how to actually connect to my container.
Have no fear! I don't need environment variables, I know that docker will expose a service in the /etc/hosts file! Sure enough I get:

    172.17.27.199 8f0c96d2e2b7
    127.0.0.1 localhost
    ::1 localhost ip6-localhost ip6-loopback
    fe00::0 ip6-localnet
    ff00::0 ip6-mcastprefix
    ff02::1 ip6-allnodes
    ff02::2 ip6-allrouters
    172.17.27.198 mycontainer 7e1d0fda4e1c wercker-service-mycontainer_latest-5605902bce3c3b07460ff35d
    172.17.27.197 postgresql b939263635e9 wercker-service-postgresql_latest-5605902bce3c3b07460ff35d

Which shows me `172.17.27.198 mycontainer`. Ok so if I curl mycontainer everything should work. But no:

    $ curl -s mycontainer:8085
    curl: (7) Failed to connect to mycontainer port 8085: Connection refused

Probably because wercker has some type of firewall/port-forwarding that needs to be touched. So long story short, I cant use wercker published containers as wercker services.

Proposed solution, "ports" on the docker push actually exposes the port I want to be exposed.

**Comment by hatchan** [comment]
Hi sorry for the late reply. 

The notation for the port section of `internal/docker-push` is actually just a string separated by comma's: 

```
ports: "5000, 8085"
```

I'll update the documentation to make note of this.

**Comment by hatchan** [comment]
Created the PR for our documentation: wercker/docs#288
